lib/internal/process.js

```diff
@@ -191,6 +191,11 @@ function setupSignalHandlers() {
   process.on('newListener', function(type, listener) {
     if (isSignal(type) &&
         !signalWraps.hasOwnProperty(type)) {
+
+      if (['SIGBUS', 'SIGFPE', 'SIGSEGV', 'SIGILL'].includes(type)) {
+        process.emitWarning('SIGBUS, SIGFPE, SIGSEGV and SIGILL are not safe');
```

Comment:
The message could probably be a bit more descriptive.

e.g. 
````
Listening to X, X, etc signals are not safe and the listener may be called in an infinite loop
```

or something